### DISCUSSION TITLE
q-input does not return the appropriate type

### DISCUSSION THREAD

**Discussion by pylvr** [quasarframework/quasar#13417] [General - Components / Directives / etc]
### Description

  say that we have a component 
 
 <q-input type="number" v-model="XXX"/>
 
 is there a way that the component can return the appropriate type based on the attribute 'type' , 
 for instance it always return 'string' 

### Documentation Section URL

https://quasar.dev/vue-components/input#introduction

### Areas

Components (quasar)

**Comment by pdanpdan** [comment]
Return whatever you want from your component, if it has a type prop and you emit the value based on that type.
If you mean something else please explain.
